src/.r-install.sh

```diff
@@ -1,178 +1,26 @@
-# except easier access functions in R for external C
-# function calls.
-#
-# For example instead of typing
-#
-#     > .External("$2", "send_command", "blah")
-#
-# one can simply type
-#
-#     > send_command(blah)
-#
-# where > is the R prompt.
-
-# interface $2(arg1,arg2,...) as w/ matlab/octave/python
-#
-$2 <- function(...) .External("$2",...,PACKAGE="$2")
-
-
-# R specific interface
-
-# Generic functions
-#
-send_command <- function(x) .External("$2","send_command",x,PACKAGE="$2")
-set_features <- function(x,y) .External("$2","set_features",x,y,PACKAGE="$2")
-add_features <- function(x,y) .External("$2","add_features",x,y,PACKAGE="$2")
-set_labels <- function(x,y) .External("$2","set_labels", x,y,PACKAGE="$2")
-get_kernel_matrix <- function() .External("$2","get_kernel_matrix",PACKAGE="$2")
-
-# SVM functions
-#
-svm_classify <- function() .External("$2","svm_classify",PACKAGE="$2")
-get_svm <- function() .External("$2","get_svm",PACKAGE="$2")
-get_subkernel_weights <- function() .External("$2","get_subkernel_weights",PACKAGE="$2")
-
-# HMM functions
-#
-get_hmm <- function() .External("$2","get_hmm",PACKAGE="$2")
-
-# Load the shogun dynamic library at startup.
-#
-.First.lib <- function(lib,pkg)
-{
-	cat(paste("\nWelcome! This is SHOGUN version $VERSION\n"))
-	library.dynam("$2",pkg,lib)
-}
-
-# Unload the library.
-#
-.Last.lib <- function(lib) library.dynam.unload("$2", libpath=lib)
-
-# Because in packages with namespaces .First.lib will not be loaded
-# one needs another functions called .onLoad resp. .onUnload
-#
-
-.onLoad <- function(lib, pkg) .First.lib(lib,pkg)
-.onUnload <- function(lib) .Last.lib(lib)
-EOF
-
-# R-MODULAR
-else
 echo "Installing modular shogun interface for R"
 
-cat >"$1/$2/NAMESPACE" <<EOF
-useDynLib(shogun, .registration = TRUE)
-EOF
-
-cat >"$1/$2/R/$2" <<EOF
-.packageName <- "$2"
-#$2 <- function(...) .External("$2",...,PACKAGE="$2")
-
-# Load the shogun dynamic library at startup.
-#
-.First.lib <- function(lib,pkg)
-{
-	cat(paste("\nWelcome! This is SHOGUN version $VERSION\n"))
-EOF
+# clean up
+rm -rf ${PACKAGE_NAME}
 
-for f in *.so
-do
-	echo "library.dynam(\"`basename $f .so`\", pkg, lib)" >> "$1/$2/R/$2"
-done
+# create package directories
+mkdir -p ${PACKAGE_NAME}/R
+mkdir -p ${PACKAGE_NAME}/inst/libs
 
-for f in *.RData
-do
-	echo "load(paste(lib, \"/\", \"$2\", \"/R/\", \"`basename $f`\", sep=''), envir=.GlobalEnv)" >> "$1/$2/R/$2"
-	echo "cacheMetaData(1)" >> "$1/$2/R/$2"
-done
-cat >>"$1/$2/R/$2" <<EOF
-}
-# Unload the library.
-#
-.Last.lib <- function(lib)
-{
-EOF
+cp *${LIB_EXTENSION} ${PACKAGE_NAME}/inst/libs/
+cp *.R ${PACKAGE_NAME}/R/
+cp DESCRIPTION ${PACKAGE_NAME}/
+cp NAMESPACE.i ${PACKAGE_NAME}/NAMESPACE
 
-for f in *.$3
-do
-	echo "library.dynam.unload(\"$f\", lib)" >> "$1/$2/R/$2"
-done
+# use full matching instead of partial matching
+sed -i 's/pmatch(name, names(accessorFuns));/match(name, names(accessorFuns));/g' ${PACKAGE_NAME}/R/*.R
```

Comment:
SWIG does partial matching for function names, which causes problems. For example, calling `kernel$add(/* some arguments*/)` will invoke `kernel$add_to_normal` instead of `CSGObject::add`, since it first searches for the classes methods, then the parent.
I can't see any other way of doing this instead of this dirty hack.